### PR TITLE
[11.0] mig operating unit

### PR DESCRIPTION
**Pull request by gdgellatly**
- Change groups_id inheritance in view to specific group as groups_id gets replaced by a same priority view.
- Remove encodings
- Bump Version
- Test
- Add known issues for docs - if you know the answer I can update docs.
- Update name_search signature to match BaseModel

### CODE REVIEW

File: operating_unit/README.rst
Comment (AaronHForgeFlow) [inline operating_unit/README.rst:58]: You can add yourself as contributor

**Comment by gdgellatly** [comment]
Review URL https://github.com/OCA/operating-unit/pull/99/commits/5592d2c8ab8fa62d2aeb89b4d67ff016ed9e910a to make it easier for reviewers

**Review by AaronHForgeFlow — APPROVED** [review]
Code seems ok to me. Functional ok also so :+1:

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
@gdgellatly  regarding the issue mentioned, the partner means to contain the operating unit information. That is, address, logo, phone all contact information that can be used to customized printouts by operating unit or provide the address to other contacts.

**Comment by gdgellatly** [comment]
@aheficent alright simple enough, I added a help message to partner id and removed the known issues

File: operating_unit/README.rst
Comment (gdgellatly) [reply operating_unit/README.rst:58]: It doesn't matter.  I don't usually unless its a substantial amount of work.

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by gustavovalverde** [comment]
As far as I tested, this seems OK for the first migration. Do you need / want further testing before making it available for v11?

**Comment by AaronHForgeFlow** [comment]
@gustavovalverde if you think this is ok to be merged you can approved this PR and then it will be merged.  Currently it has 3 approvals but all of them are from @eficent team

**Review by tschanzt — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
@gdgellatly there are too many commits from older versions, can you squash those a little bit?

**Comment by gdgellatly** [comment]
@aheficent Nope, it keeps failing on rebase.

File: operating_unit/README.rst
Comment (simahawk) [inline operating_unit/README.rst]: `Settings / Users & Companies / Operating Units`

File: operating_unit/README.rst
Comment (simahawk) [inline operating_unit/README.rst]: /11.0

File: operating_unit/__init__.py
Comment (simahawk) [inline operating_unit/__init__.py]: replace :copyright: w/ "Copyright"

File: operating_unit/__manifest__.py
Comment (simahawk) [inline operating_unit/__manifest__.py]: https://github.com/OCA/operating-unit

File: operating_unit/models/__init__.py
Comment (simahawk) [inline operating_unit/models/__init__.py]: remove coding from *.py

File: operating_unit/models/operating_unit.py
Comment (simahawk) [inline operating_unit/models/operating_unit.py:17]: can we avoid splitting the lambda? keyword args can stay each on one line :)

File: operating_unit/models/operating_unit.py
Comment (simahawk) [inline operating_unit/models/operating_unit.py]: `_` translation missing

File: operating_unit/models/res_users.py
Comment (simahawk) [inline operating_unit/models/res_users.py]:
```python
user = self.env.user
if uid2:
    user = self.env['res.users'].browse(uid2)
```

File: operating_unit/models/res_users.py
Comment (simahawk) [inline operating_unit/models/res_users.py]: isn't this the same as passing nothing?

File: operating_unit/models/res_users.py
Comment (simahawk) [inline operating_unit/models/res_users.py]: use a lambda to proxy the method call instead of referencing the method's object:
```python
default=lambda self: self._default_operating_units()
[...]
default=lambda self: self._default_operating_unit()
```
Note the change in the name of the methods ;)

File: operating_unit/security/operating_unit_security.xml
Comment (simahawk) [inline operating_unit/security/operating_unit_security.xml]: no need for list comprehension, use `.ids`: `[('id','in', user.operating_unit_ids.ids])]`

File: operating_unit/view/operating_unit_view.xml
Comment (simahawk) [inline operating_unit/view/operating_unit_view.xml]: name groups, for the sake of future extension (when you'll add a new group you will copy paste it w/ a name attr ;))

**Review by simahawk — CHANGES_REQUESTED** [review]

File: operating_unit/models/res_users.py
Comment (gdgellatly) [reply operating_unit/models/res_users.py]: @simahawk Honestly I don't know, not the original author and I just couldn't understand why it was like it was.  Plus I've not looked at this in months. But it is different, passing nothing would give an error.  

I think I changed most of this file, then reverted it because other modules used this code.  

But in practice I've found these methods are unsuitable anyway and have rewritten because the only time you really care about a users default operating unit is in creation of a business document, and then you get all sorts of weird issues (like in a sales order, the warehouse,  Sales Team, salesperson and the user entering might all belong to different OU's and in multicompany it is a nightmare).

EDIT: I guess what I'm saying is I don't even know why these methods exist, especially the plural one.

File: operating_unit/models/operating_unit.py
Comment (ghost) [inline operating_unit/models/operating_unit.py:33]: calls to `super()` no longer require arguments as of Python3 🙂

**Review by ghost — APPROVED** [review]

File: operating_unit/models/res_users.py
Comment (simahawk) [reply operating_unit/models/res_users.py]: @gdgellatly we are already doing it https://github.com/OCA/operating-unit/blob/10.0/sale_operating_unit/models/sale.py#L14
This is implementation is really old, there's no reason to keep it as it is. 
The ones that rely on this old way will have to migrate modules to v11 as well. 
So, this is the right time to make this change ;)

Same goes for the indirection of default methods: ppl used to add 2 methods to make them overridable but if you use lambda to proxy the call you don't need the double step anymore.

**Review by edser-pcg — APPROVED** [review]

**Review by edser-pcg — APPROVED** [review]

**Review by edser-pcg — APPROVED** [review]

**Comment by edser-pcg** [comment]
When will this be merged?

**Comment by elicoidal** [comment]
@simahawk @gdgellatly anything that can be added to improve the original PR or shall we add some FIXME in the code?

**Comment by simahawk** [comment]
@elicoidal to me those methods should be fixed now before merge. Is not a big change BTW.
I'm not blocking tho, if we put some `#FIXME` comment :wink:

**Comment by gdgellatly** [comment]
I'll fix them. Just got new laptop so it takes a while to set up.

**Comment by gdgellatly** [comment]
is CLA a me or you issue?

**Comment by AaronHForgeFlow** [comment]
@gdgellatly the problem is the last commit I think. The email address of the author it's different as the one it's registered. It will be solved by amending the last commit and changing the author.

**Comment by gdgellatly** [comment]
@simahawk happy now :).  btw, ignore the CLA message, I force committed after.  New laptop had wrong git config.

**Comment by JordiBForgeFlow** [comment]
Can you rebase? I am not able to merge.

**Comment by gdgellatly** [comment]
@jbeficent I can try, but should be nothing to rebase.

**Comment by JordiBForgeFlow** [comment]
I am getting a message:
This branch cannot be rebased safely
Rebasing the commits of this branch on top of the base branch cannot be performed automatically as this would create a different result than a regular merge.

**Comment by gdgellatly** [comment]
@jbeficent don't worry, new laptop issues, had remotes around wrong way, will fix up now

**Comment by gdgellatly** [comment]
@simahawk @jbeficent Wow I made some messes for myself.  Had been force pushing to this repo on 2 seperate branches mig-operating-unit and mig-operating_unit and then through some magic these were merging into each other.  And then my upstream repo wasn't oca but a personal one.  I seem to recall why this arrangement made sense when I started this, but months later forgot I did it.

So what I did was copy the files from the last branch I was working on which was what was reviewed.
Delete original branches.
Restarted migration process from scratch.
Copy files back
Commit
Force push.

So please just give it one final check and make sure I've not messed/missed anything in this process.  I've given it a run through and it seems fine.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: From my side this is ready to merge.

File: operating_unit/models/operating_unit.py
Comment (simahawk) [inline operating_unit/models/operating_unit.py:23]: make this translatable?

File: operating_unit/models/operating_unit.py
Comment (simahawk) [reply operating_unit/models/operating_unit.py:17]: ping :)

**Review by simahawk — CHANGES_REQUESTED** [review]
@gdgellatly seems good. Just 2 small comments. Most important: translation of sql constraints.

**Review by edser-pcg — APPROVED** [review]
All seems good.

**Pull request merged**
